### PR TITLE
Gourmonger Tweaks: The Second Course

### PR DESCRIPTION
**Pull request by GreatBigFailure**
Gourmongers were a little too hard to manage as they were. The success/failure state was too loose and hard to control. Additionally if you avoided feeding them until splitting they would be incredibly easy to keep satisfied (100u nutriment keeps them happy for over an hour) while babies would almost always go feral right after being born. This fixes that, hopefully.

**Major Changes**

- Gourmonger splitting was always based on amount of meals, not nutrition of those meals. Now instead of just having a tally they keep forever Gourmongers will lose progress toward splitting over time. The closer they are to splitting the higher the chance of losing a meal point. Maintaining reasonable population is now more achievable.
- Gourmongers previously lost 1 nutrition a tick. That meant a single pizza was good for more than 10 minutes of not loosing. Gourmongers now lose increasingly more nutrition the longer its been since their last meal. This is reset when they lose a meal point, meaning a close-to-split Gourmonger will stay full longer.
- When splitting Gourmongers would feed their fresh, near-starving kids by exploding into meat chunks. They still do but they also directly transfer an additional 10% of their nutrition to their kids. Not only making them not immediately feral but effectively acting as a multiplier when combined with the meat.
- Gourmongers now glow more intensely the closer they are to splitting. When 1-2 meals away they will be very obvious.

**Minor Changes**

- Gourmongers now run slower when hungry. They'll still catch you if you aren't careful, but they are no longer chubby green sanics. 
- Their knockdown on impact has also been reduced to less than half of what it was. You will now stand up before they're able to knock you down again. You'll be slowed from the pain but still able to fight back.
- Instead of releasing a huge rad burst on death they now release it on splitting.
- Examining a Gourmonger tells you they're close to starvation if within 100 nutrition of starving, and tells you they're very fat if above 1000 nutrition.
- Gourmongers gain 1 max health every meal. Possibly meaning the difference between 3 shots to kill and 4. Unpredictable!
- Number tweaks: Shorter damage (not engine) rad range, feeding them humans is now more nutrition-profitable, etc

[tweak]

:cl:
 * tweak: Gourmonger behavior/mechanic/number tweaks.

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
I'll write the wiki for them tomorrow. Hopefully that helps the confusion about their mechanics.

**Comment by Eneocho** [comment]
> 
> 
> I'll write the wiki for them tomorrow. Hopefully that helps the confusion about their mechanics.

I created a wiki page for gours yesterday, but due to the volatile nature of the recent merges I haven't added anything noteworthy.

**Comment by GreatBigFailure** [comment]
> I created a wiki page for gours yesterday, but due to the volatile nature of the recent merges I haven't added anything noteworthy.

Oh Fantastic, thanks. I'll add to it today then.

**Comment by Zth--** [comment]
```
[09:50:30] Runtime in gourmonger.dm,228: Cannot read null.loc
  proc name: chargeToPrey (/mob/living/simple_animal/hostile/gourmonger/proc/chargeToPrey)
  src: Gourmonger (/mob/living/simple_animal/hostile/gourmonger)
  src.loc: the floor (125,65,2) (/turf/simulated/floor)
  call stack:
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): chargeToPrey(null)
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): Goto(null, 2, 1)
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): MoveToTarget()
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): Life()
  ```


```
[09:41:24] Runtime in gourmonger.dm,240: Cannot read null.contents
  proc name: chargeThrough (/mob/living/simple_animal/hostile/gourmonger/proc/chargeThrough)
  src: Gourmonger (/mob/living/simple_animal/hostile/gourmonger)
  src.loc: null
  call stack:
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): chargeThrough(null)
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): chargeToPrey(Homer O\'Brien (/mob/living/carbon/human))
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): findPrey()
```


```
[09:56:55] Runtime in gourmonger.dm,122: Division by zero
  proc name: divideMeat (/mob/living/simple_animal/hostile/gourmonger/proc/divideMeat)
  src: Gourmonger (/mob/living/simple_animal/hostile/gourmonger)
  src.loc: the floor (239,292,1) (/turf/simulated/floor)
  call stack:
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): divideMeat()
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): death()
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): adjustBruteLoss(2)
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): adjustBruteLoss(2)
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): handle environment(/datum/gas_mixture (/datum/gas_mixture))
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): Life()
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): Life()
  Gourmonger (/mob/living/simple_animal/hostile/gourmonger): Life()
  Mob (/datum/subsystem/mob): fire(0)
  Mob (/datum/subsystem/mob): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing()
```

File: code/modules/power/gourmonger.dm [hunk 1]
```diff
@@ -17,45 +16,54 @@ var/global/gourmonger_saturation = 0
 	maxHealth = 125
 	melee_damage_lower = 5
 	melee_damage_upper = 10	//The rads are the real danger
-	stat_attack = 2
-	search_objects = 1
+	stat_attack = 2	//So it attacks corpses.
```
Comment (ShiftyRail) [inline code/modules/power/gourmonger.dm]: Replace this with the define, while you're at it.
```suggestion
	stat_attack = DEAD //So it attacks corpses.
```

**Review by ShiftyRail — APPROVED** [review]

**Comment by GreatBigFailure** [comment]
(Maybe) fixed the runtimes, some minor tweaking, and I noticed they act really weird if you put them in a cage so I fixed that too.

**Comment by Eneocho** [comment]
Do gours smash through reinforced walls?

**Comment by GreatBigFailure** [comment]
> Do gours smash through reinforced walls?

Yes

**Pull request merged**